registration/src/main/java/org/spine3/samples/lobby/registration/seat/availability/SeatsAvailabilityAggregate.java

```diff
@@ -0,0 +1,366 @@
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.samples.lobby.registration.seat.availability;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Message;
+import org.spine3.base.CommandContext;
+import org.spine3.samples.lobby.common.ReservationId;
+import org.spine3.samples.lobby.common.SeatTypeId;
+import org.spine3.samples.lobby.registration.contracts.SeatQuantity;
+import org.spine3.server.Assign;
+import org.spine3.server.aggregate.Aggregate;
+import org.spine3.server.aggregate.Apply;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Lists.newLinkedList;
+import static org.spine3.samples.lobby.registration.util.CollectionUtils.findById;
+import static org.spine3.samples.lobby.registration.util.MessageFactory.newSeatQuantities;
+import static org.spine3.samples.lobby.registration.util.MessageFactory.newSeatQuantity;
+import static org.spine3.samples.lobby.registration.util.ValidationUtils.*;
+
+/**
+ * The aggregate which manages the availability of conference seats.
+ *
+ * @author Alexander Litus
+ */
+@SuppressWarnings({"TypeMayBeWeakened"/** "OrBuilder" parameters are not applicable*/, "OverlyCoupledClass"})
+public class SeatsAvailabilityAggregate extends Aggregate<SeatsAvailabilityId, SeatsAvailability> {
+
+    /**
+     * Creates a new instance.
+     *
+     * @param id the ID for the new instance
+     * @throws IllegalArgumentException if the ID type is not supported
+     */
+    public SeatsAvailabilityAggregate(SeatsAvailabilityId id) {
+        super(id);
+    }
+
+    @Override
+    protected SeatsAvailability getDefaultState() {
+        return SeatsAvailability.getDefaultInstance();
+    }
+
+    @Assign
+    public SeatsReserved handle(MakeSeatReservation cmd, CommandContext context) {
+        Validator.validateCommand(cmd);
```

Comment:
Do you mean moving all methods from Validator to its parent? But then a warning "Class with too  many methods" will appear. And it is good to have a separate class for this. We can rename "validateCommand" to "check" and "validateState" to "checkState", that would be simpler.